jmclient/jmclient/blockchaininterface.py

```diff
@@ -285,18 +285,34 @@ def import_addresses_if_needed(self, addresses, wallet_name):
             self.import_addresses(addresses - imported_addresses, wallet_name)
         return import_needed
 
-    def _yield_transactions(self, wallet_name):
-        batch_size = 1000
-        iteration = 0
+    def _yield_transactions(self, wallet_name=None):
```

Comment:
My first question is, why is there a  change to `_yield_transactions` if you're trying to improve the efficiency of the monitor loop? Did I miss something or isn't this only used in wallet initial sync? (Technically it is also used in one place to do with SNICKER, but no one's using that).

Ah is it specifically that you wanted to make a `yield_new_transactions` functions leveraged off this? OK. It will mean, though, very carefully examining to make sure there's no functional change to wallet sync.